tools/serverpod_cli/lib/src/generator/open_api/objects/schema.dart

```diff
@@ -0,0 +1,199 @@
+// ignore_for_file: public_member_api_docs, sort_constructors_first
+part of '../open_api_objects.dart';
+
+/// Serializes [TypeDefinition] all type To toJson.
+Map<String, dynamic> typeDefinitionToJson(TypeDefinition type,
+    [bool child = false]) {
+  Map<String, dynamic> map = {};
+  if (type.className == 'Future') {
+    return typeDefinitionToJson(type.generics.first, child);
+  }
+  if (type.className == 'void') return map;
+  if (type.isMapType) {
+    map = mapTypeToJson(type, child);
+    return map;
+  }
+  if (type.isListType) {
+    map = listTypeToJson(type, child);
+    return map;
+  }
+
+  if (type.isOtherType) {
+    map = otherTypeToJson(type, child);
+    return map;
+  }
+  map = dartCoreTypeToJson(type, child);
+  return map;
+}
+
+/// Serializes a [TypeDefinition]  [Map] type to Json
+Map<String, dynamic> mapTypeToJson(TypeDefinition type, [bool child = false]) {
+  assert(type.isMapType,
+      'Use mapToJson only when the typeDefinition is of the MapType.');
+
+  Map<String, dynamic> map = {};
+  map['type'] = SchemaObjectType.object.name;
+  if (type.nullable) map['nullable'] = true;
+
+  // If data is Map<String,int>  use last int from generics as
+  // additionalProperties
+  if (type.generics.isEmpty) return map;
+  var lastType = type.generics.last;
+  if (lastType.isListType) {
+    map['additionalProperties'] = listTypeToJson(lastType, true);
+  } else if (lastType.isMapType) {
+    map['additionalProperties'] = mapTypeToJson(lastType, true);
+  } else if (lastType.isOtherType) {
+    map['additionalProperties'] = otherTypeToJson(lastType, true);
+  } else {
+    map['additionalProperties'] = dartCoreTypeToJson(
+      lastType,
+      true,
+    );
+  }
+
+  return map;
+}
+
+/// Serializes a [TypeDefinition]  dart core type (string,bool,double,int,
```

Comment:
Mind spacing. Here there is an extra space after `[TypeDefinition]`, there should be spaces after the commas. The correct name for `String` should have an upper case `S`.